packages/edit-post/src/components/layout/style.scss

```diff
@@ -103,6 +103,13 @@
 		@supports (position: sticky) {
 			flex-basis: 100%;
 		}
+
+		// This is an ugly hack that applies a table display to the main editing canvas when the browser is
+		// detected to be in quirks mode. Without this hack in quirks mode, the flexbox rules we use to size
+		// the various elements do not work, causing overlap.
+		.is-quirks-mode & {
+			display: table;
```

Comment:
Why do the whole detecting quirks mode song and dance, if this simple rule "fixes" the issue?

Partially because we should not optimize our CSS to be able to run in quirks mode. While we're currently primarily seeing the effects on the main editing canvas, quirks mode is likely the cause of a plethora of other tiny visual regressions, and instead of having users report those as Gutenberg issues that are really hard to reproduce and debug, it is probably a better idea to fix the issue at the root: ensure that Gutenberg loads in standards mode.

One other issue that has been reported, is that editor styles that apply a background color, that color does not extend past the initial height of the viewport.

Secondly, because `display: table;` is a heavy handed rule that is likely to break in a bunch of other ways. The reason it "works" is because an element that behaves as a table will _always_ expand to contain the contents inside. But this behavior itself is also what makes it likely to cause other issues.